rskj-core/src/main/java/co/rsk/peg/ErpFederation.java

```diff
@@ -111,60 +54,20 @@ public Script getStandardP2SHScript() {
         return standardP2SHScript;
     }
 
-    @Override
-    public Address getAddress() {
-        if (address == null) {
-            address = Address.fromP2SHScript(btcParams, getP2SHScript());
-        }
-
-        return address;
-    }
-
-    @Override
-    public boolean equals(Object other) {
-        if (this == other) {
-            return true;
-        }
-
-        if (other == null || this.getClass() != other.getClass()) {
-            return false;
+    private void validateErpFederationValues() {
+        if (erpPubKeys == null) {
+            String message = "Provided erpPubKeys is empty";
+            throw new VerificationException(message);
         }
 
-        ErpFederation otherErpFederation = (ErpFederation) other;
-
-        return this.getNumberOfSignaturesRequired() == otherErpFederation.getNumberOfSignaturesRequired() &&
-            this.getSize() == otherErpFederation.getSize() &&
-            this.getCreationTime().equals(otherErpFederation.getCreationTime()) &&
-            this.creationBlockNumber == otherErpFederation.creationBlockNumber &&
-            this.btcParams.equals(otherErpFederation.btcParams) &&
-            this.members.equals(otherErpFederation.members) &&
-            this.getRedeemScript().equals(otherErpFederation.getRedeemScript()) &&
-            this.erpPubKeys.equals(otherErpFederation.erpPubKeys) &&
-            this.activationDelay == otherErpFederation.activationDelay;
-    }
-
-    @Override
-    public int hashCode() {
-        // Can use java.util.Objects.hash since all of Instant, int and List<BtcECKey> have
-        // well-defined hashCode()s
-        return Objects.hash(
-            getCreationTime(),
-            this.creationBlockNumber,
-            getNumberOfSignaturesRequired(),
-            getBtcPublicKeys(),
-            getErpPubKeys(),
-            getActivationDelay()
-        );
-    }
-
-    private void validateRedeemScript() {
-        if (activations.isActive(ConsensusRule.RSKIP293) &&
-            this.redeemScript.equals(new Script(ERP_TESTNET_REDEEM_SCRIPT_BYTES))) {
-
-            String message = "Unable to create ERP Federation. The obtained redeem script matches the one hardcoded for testnet. "
-                + "This would cause bitcoinj-thin to identify it as invalid";
-            logger.debug("[validateRedeemScript] {}", message);
-            throw new FederationCreationException(message);
+        long maxCsvValue = ErpFederationRedeemScriptParser.MAX_CSV_VALUE;
+        if (activationDelay <= 0 || activationDelay > maxCsvValue) {
+            String message = String.format(
+                "Provided csv value %d must be larger than 0 and lower than %d",
+                activationDelay,
+                maxCsvValue
+            );
+            throw new VerificationException(message);
```

Comment:
```suggestion
            throw new FederationCreationException(message);
```
That one seems more appropiate, `VerificationException` is a bitcoinj-thin exception